### ISSUE TITLE
wavesurfer.js will lose the ability to setTime, after the timestamp is set to a specific value.

### ISSUE THREAD

**Issue by Moskize91** [katspaugh/wavesurfer.js#3823]
## Bug description
use this wav file ([data/record.wav](https://github.com/Moskize91/__wavesurfer_seeking_error/blob/main/data/record.wav)) I prepared, it is special.

when `wavesurfer.setTime()` is called to specify a specific time (300 seconds in this case), the wavesurfer.js instance will lose the ability to correctly set `setTime` to the specified time. In fact, no matter what value is set by `setTime` after this, the wavesurfer player will go to the 0 second position.

I program in TypeScript and use the react version of wavesurfer. I have put together a repo with the minimum set of bugs that can stably reproduce the bug (https://github.com/Moskize91/__wavesurfer_seeking_error). I have prepared a script that you should be able to launch and reproduce the bug easily (in linux or macOS).

## Environment
 - Browser: Chrome 127.0.6533.100 (arm64)
 - OS: macOS 14.5 (23F79)

## Minimal code snippet
https://github.com/Moskize91/__wavesurfer_seeking_error

once you have started, use your browser to access http://localhost:8080/. Then click the button to start debugging. before doing this, remember to open the console and observe the log. you will see that the last `setTime` operation caused the jump to not work properly.

### main code to reproduce the bug
https://github.com/Moskize91/__wavesurfer_seeking_error/blob/69d18bff11dc0a06ca2e2f371aa5c23e6e09ea26/src/index.tsx#L41-L65
```typescript
    const wrongTime = 300.0;
    const expectTime = 150.0;


    await sleep(100);


    // this step is Ok
    wavesurfer.setTime(100);
    wavesurfer.play();
    await sleep(2500);


    // this step is Ok.
    // but this timestamp will cause the next step to be wrong.
    wavesurfer.setTime(wrongTime);
    await sleep(1500);


    // this step isn't Ok.
    // if you comment out the previous step, this step will work fine.
    wavesurfer.setTime(expectTime);


    await sleep(0);
    const currentTime = wavesurfer.getCurrentTime();


    console.log("expectTime:", expectTime); // 150.0
    console.log("currentTime:", currentTime); // 0.0
    console.log("debug is done. please click stop button to stop the audio.");
```

## Expected result
the last `setTime` can successfully go to the 150 second

## Obtained result
go to the 0 second

## Screenshots
<img width="1265" alt="截屏2024-08-14 10 10 36" src="https://github.com/user-attachments/assets/7fe5a1cb-81a1-4098-948d-d047510a58d0">

**Comment by katspaugh** [comment]
Thanks for the report! Any idea what causes this? What’s special about this audio file apart from the length? Is it VBR by any chance?

**Comment by Moskize91** [comment]
I replaced the wavesurfer.js instance with HTMLAudioElement and the problem still exists. You can refer to the new file `audio.tsx` I wrote.

> https://github.com/Moskize91/__wavesurfer_seeking_error/blob/main/src/audio.tsx

so it seems that it is not a problem with wavesurfer.js, but that the browser player cannot handle this file correctly. I will investigate what happens in ffmpeg and youtube video downloads, and if I find anything, I will say it in this issue.

**Comment by katspaugh** [comment]
Great, thanks! I'll close the issue then but please do write here if you find any workaround we can add.

**Comment by Moskize91** [comment]
everything is clear now. ehe problem is not with wavesurfer.js or the browser audio player. It is with the `vite.config.ts` file I wrote.

https://github.com/Moskize91/__wavesurfer_seeking_error/blob/3f7bcdb50c32ed3b6b558a16540fc4a35dce7d20/vite.config.ts#L16-L25
```javascript
        if (req.url && req.url.startsWith("/data/")) {
          const filePath = path.resolve(__dirname, "." + req.url!);
          if (fs.existsSync(filePath)) {
            const fileStream = fs.createReadStream(filePath);
            res.writeHead(200, { "Content-Type": "application/octet-stream" });
            fileStream.pipe(res);
          } else {
            res.statusCode = 404;
            res.end('Not Found');
          }
```

actually, I mistakenly thought that wavesurfer.js would download the entire audio file. however, the behavior of wavesurfer.js is the behavior of the browser player (wavesurfer.js directly uses the browser player). therefore, when the audio length exceeds 5 minutes, the browser player will have a "range" field in the request headers. I did not handle it in `vite.config.ts`. this caused the browser player to behave abnormally.

in fact, if the browser sends a request with a range field, the server should return 206 with the Content-Range field in the headers. but even if I didn't do this, the browser console didn't report an error or throw a warning, which is really strange.

however, this has nothing to do with wavesurfer.js, it doesn't need to handle these things. because most users will save audio files in something like AWS S3, and it will handle the range field correctly. they won't encounter this problem when using wavesurfer.js.

that's it for this question, thanks a lot to @katspaugh katspaugh  for keeping an eye on this issue.
